ably/realtime_conn.go

```diff
@@ -86,23 +90,78 @@ func newConn(opts *clientOptions, auth *Auth, callbacks connCallbacks) (*Connect
 		callbacks: callbacks,
 	}
 	c.queue = newMsgQueue(c)
+	c.ctx, c.cancel = context.WithCancel(context.Background())
+	go c.watchDisconnect(c.ctx)
 	if !opts.NoConnect {
-		if _, err := c.connect(false); err != nil {
+		if _, err := c.connect(connArgs{}); err != nil {
 			return nil, err
 		}
 	}
 	return c, nil
 }
 
-func (c *Connection) dial(proto string, u *url.URL) (proto.Conn, error) {
+func (c *Connection) dial(proto string, u *url.URL, once bool) (conn proto.Conn, err error) {
+	lg := c.logger().Sugar()
+	start := time.Now()
+	lg.Debugf("Dial protocol=%q url %q ", proto, u.String())
 	// (RTN23b)
 	query := u.Query()
 	query.Add("heartbeats", "true")
 	u.RawQuery = query.Encode()
+	// We first try a single dial to see if we are successful. There is no need to
+	// have this only in the for loop because most cases we will succeed after the
+	// first attempt, let's be prudent here.
+	conn, err = c.dialInternal(proto, u, c.opts.realtimeRequestTimeout())
```

Comment:
I think this should be using [`httpOpenTimeout`](https://www.ably.io/documentation/client-lib-development-guide/features#TO3l3)

Reply:
done 01b4ab2f6e0a0c19d7da4bd465dcc42ab6b2f810